File: leetcode/Python/0080-m-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        newTail = 0
        times = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[newTail]:
                newTail += 1
                nums[newTail] = nums[i]
                times = 1
            else:
                times += 1
                if times <= 2:
                    newTail += 1
                    nums[newTail] = nums[i]
        return newTail + 1

File: leetcode/C++/0080-m-remove-duplicates-from-sorted-array-ii.cpp
// Time: O(n)
// Space: O(1)
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int cur = 0;
        for (int i = 0; i < n; ++i) {
            nums[cur++] = nums[i];
            if (cur >= 3 && nums[cur-1] == nums[cur-2] && nums[cur-2] == nums[cur-3])
                cur--;
        }
        return cur;
    }
};